File: be/src/main/java/sg/flow/services/AuthServices/FlowTokenService.java
package sg.flow.services.AuthServices;

import java.util.Optional;

import sg.flow.models.auth.FlowUserDetails;
import sg.flow.models.auth.TokenSet;

public interface FlowTokenService {

    Optional<FlowUserDetails> getUserDetailByAccessToken(String accessToken);

    Optional<TokenSet> getAccessTokenByRefreshToken(String refreshToken);

    Optional<TokenSet> generateAndStoreAccessToken(String refreshToken);

    Optional<TokenSet> generateAndStoreRefreshToken(Integer userId);

}

File: be/src/main/kotlin/sg/flow/services/AuthServices/FlowTokenService.kt
package sg.flow.services.AuthServices

import com.nimbusds.oauth2.sdk.token.AccessToken
import sg.flow.models.auth.FlowUserDetails
import sg.flow.models.auth.TokenSet
import javax.accessibility.Accessible

interface FlowTokenService {
    suspend fun getUserDetailByAccessToken(accessToken: String): FlowUserDetails?
    suspend fun getAccessTokenByRefreshToken(refreshToken: String): TokenSet?
    suspend fun generateAndStoreAccessToken(refreshToken: String): TokenSet?
    suspend fun generateAndStoreRefreshTokenAndAccessToken(userId: Int): TokenSet?
    suspend fun revokeAccessToken(accessToken: String): Boolean
}
